### DISCUSSION TITLE
SentimentAnalysis

### DISCUSSION THREAD

**Discussion by meyrow** [deepjavalibrary/djl#1321] [Development]
Hello
I would like to use SentimentAnalysis example. 
Unfortunally the code failed with ModelNotFoundException

> Task :examples:SentimentAnalysis.main()
[INFO ] - input Sentence: I like DJL. DJL is the best DL framework!
Exception in thread "main" ai.djl.repository.zoo.ModelNotFoundException: No matching model with specified Input/Output type found.
	at ai.djl.repository.zoo.Criteria.loadModel(Criteria.java:178)
	at ai.djl.examples.inference.SentimentAnalysis.predict(SentimentAnalysis.java:64)
	at ai.djl.examples.inference.SentimentAnalysis.main(SentimentAnalysis.java:45)

> Task :examples:SentimentAnalysis.main() FAILED

Does the example works for others?

**Comment by meyrow** [comment]
The example "ObjectDetection" works fine. Why is the example SentimentAnalysis  not working?
I debug the code and found that the reason it throws an exception is in Criteria.java:178. It is because that the code doesn`t reach line 166 "return loader.loadModel(this);"
I am interested in Sentiment Analysis so it will be helpful to have a working example.

**Comment by frankfliu** [comment]
@meyrow, The SentimentAnalysis example using a PyTorch model, we don't have MXNet model in model zoo yet. That's why you see the error. You need use PyTorch engine to run it.

Please see: https://github.com/deepjavalibrary/djl/blob/master/examples/docs/sentiment_analysis.md

**Comment by meyrow** [comment]
![image](https://user-images.githubusercontent.com/45011444/138735151-3695fde1-a542-45a2-8386-7085796f741f.png)
I am using Intellij IDE
please guide me with the run configuration. If it is possible?

**Comment by frankfliu** [comment]
@meyrow 
IntelliJ only parse build.gradle once on loading, so system property won't take effect in IntelliJ. What you need to do is change build.gradle file, and refresh the gradle project:

Change:
```
    if (System.getProperty("ai.djl.default_engine") == "PyTorch") {
        runtimeOnly project(":engines:pytorch:pytorch-model-zoo")
        runtimeOnly "ai.djl.pytorch:pytorch-native-auto:${pytorch_native_version}"
    } else if (System.getProperty("ai.djl.default_engine") == "TensorFlow") {
        runtimeOnly "ai.djl.tensorflow:tensorflow-native-auto:${tensorflow_version}"
        runtimeOnly project(":engines:tensorflow:tensorflow-model-zoo")
    } else {
        runtimeOnly project(":engines:mxnet:mxnet-model-zoo")
        runtimeOnly "ai.djl.mxnet:mxnet-native-auto:${mxnet_version}"
    }
```

to:

```
        runtimeOnly project(":engines:pytorch:pytorch-model-zoo")
        runtimeOnly "ai.djl.pytorch:pytorch-native-auto:${pytorch_native_version}"

```

**Reply by meyrow** [reply]
perfect. If works
